src/python/pants/engine/legacy/structs.py

```diff
@@ -349,34 +340,36 @@ def _filespec_for_exclude(raw_exclude, spec_path):
 
   @property
   @abstractmethod
-  def path_globs_kwarg(self):
+  def path_globs_kwarg(self) -> str:
     """The name of the `PathGlobs` parameter corresponding to this BaseGlobs instance."""
 
   @property
   @abstractmethod
-  def legacy_globs_class(self):
+  def legacy_globs_class(self) -> Type[wrapped_globs.FilesetRelPathWrapper]:
     """The corresponding `wrapped_globs` class for this BaseGlobs."""
 
-  def __init__(self, *patterns, **kwargs):
+  def __init__(
+    self, *patterns: str, spec_path: str, exclude: Optional[List[str]] = None, **kwargs,
+  ) -> None:
     self._patterns = patterns
-    self._kwargs = kwargs
-    raw_spec_path = kwargs.pop('spec_path')
-    self._file_globs = self.legacy_globs_class.to_filespec(patterns).get('globs', [])
-    raw_exclude = kwargs.pop('exclude', [])
-    self._excluded_file_globs = self._filespec_for_exclude(raw_exclude, raw_spec_path).get('globs', [])
-    self._spec_path = raw_spec_path
-
+    self._spec_path = spec_path
+    self._raw_exclude = exclude
     if kwargs:
-      raise ValueError('kwargs not supported for {}. Got: {}'.format(type(self), kwargs))
+      raise ValueError(f'kwargs not supported. Got: {kwargs}')
```

Comment:
It doesn't matter what type of `BaseGlobs` this is. We never allow kwargs, outside of `exclude`.